crypto/rsa/rsa_backend.c

```diff
@@ -96,25 +130,121 @@ int ossl_rsa_fromdata(RSA *rsa, const OSSL_PARAM params[], int include_private)
                                 ossl_rsa_mp_coeff_names))
             goto err;
 
-        /* It's ok if this private key just has n, e and d */
+        if (derive_from_pq && sk_BIGNUM_num(exps) == 0
```

Comment:
While a refactor might be warranted here, I am concerned that it won't result in more readable code for this PR.  I assert that because the refactoring ossl_rsa_fromdata will not just isolate the derive_from_pq cases in this code, but will impact uses of this function independent of that.  It would be better to address a refactor of this function in a separate PR